functions/all_words.py

```
@@ -532,3 +534,156 @@ def smush_solver(center, outer_uses, spicy, first_word, words, list_len=400,
+
+    Smush's PERFECT bonus (final score x2) requires playing the pangram first
+    AND a clean plate, so the search first tries to build a complete plan
+    around an affordable pangram. Only a fully flat result keeps the pangram
+    seed - a pangram isn't worth giving up the clean plate for - and any
+    pangram that makes the plan is sorted to the front.
+
+    Returns (plan, leftover): plan is a list of result dicts (pangram first,
+    then points desc) and leftover maps each letter the plan fails to flatten
+    to its stranded uses ({} means a complete smush).
+    """
+    norm = {str(l).lower(): int(n) for l, n in outer_uses.items()}
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)